frontend/taipy-gui/src/components/Taipy/Metric.spec.tsx

```diff
@@ -12,20 +12,321 @@
+
+    it("processes type and threshold props correctly when type is not linear", async () => {
+        render(<Metric type="angular" threshold={50} testId="test-id" />);
+        await waitFor(() => {
+            const elt = document.querySelector(".angular");
+            expect(elt).toBeInTheDocument();
+        });
+    });
+
+    it("applies style correctly when height is undefined", async () => {
+        render(<Metric testId="test-id" />);
+        await waitFor(() => {
+            const elt = document.querySelector(".js-plotly-plot");
+            expect(elt).toHaveStyle({
+                width: "100%",
+                position: "relative",
+                display: "inline-block",
+            });
+        });
+    });
+
+    it("processes type prop correctly when type is none", async () => {
+        render(<Metric type="none" testId="test-id" />);
+        await waitFor(() => {
+            const angularElm = document.querySelector(".angular");
+            const angularAxis = document.querySelector(".angularaxis");
+            expect(angularElm).not.toBeInTheDocument();
+            expect(angularAxis).not.toBeInTheDocument();
+        });
+    });
+
+    it("processes template_Dark_ prop correctly when theme is dark", async () => {
+        const darkTheme = createTheme({
+            palette: {
+                mode: "dark",
+            },
+        });
+
+        render(
+            <ThemeProvider theme={darkTheme}>
+                <Metric template_Dark_={JSON.stringify(darkTemplate)} testId="test-id" />
+            </ThemeProvider>,
+        );
+        await waitFor(() => {
+            const elt = document.querySelector(".main-svg");
+            expect(elt).toHaveStyle({
+                backgroundColor: "rgb(31, 47, 68)",
+            });
+        });
+    });
+
+    it("processes template_Light_ prop correctly when theme is not dark", async () => {
+        const lightTheme = createTheme({
+            palette: {
+                mode: "light",
+            },
+        });
+
+        render(
+            <ThemeProvider theme={lightTheme}>
+                <Metric template_Light_={JSON.stringify(lightTemplate)} testId="test-id" />
+            </ThemeProvider>,
+        );
+        await waitFor(() => {
+            const elt = document.querySelector(".main-svg");
+            expect(elt).toHaveStyle({
+                backgroundColor: "rgb(255, 255, 255)",
+            });
+        });
+    });
+
+    it.skip("logs an error when template_Dark_ prop is not a valid JSON string", () => {
+        const consoleSpy = jest.spyOn(console, "info");
+        render(<Metric template_Dark_="not a valid JSON string" testId="test-id" />);
+        expect(consoleSpy).toHaveBeenCalledWith(expect.stringContaining("Error while parsing Metric.template"));
+        consoleSpy.mockRestore();
+    }); // TODO: Not working at the moment, need to fix
+
+    it("logs an error when template_Light_ prop is not a valid JSON string", () => {
```

Comment:
is this one working ?

Reply:
The Light is working but Dark one it not, I don't know why yet so I skip it at the moment